src/Dictionaries/TrieDictionary.h

```diff
@@ -150,9 +148,21 @@ class TrieDictionary final : public IDictionaryBase
     BlockInputStreamPtr getBlockInputStream(const Names & column_names, size_t max_block_size) const override;
 
 private:
+
     template <typename Value>
     using ContainerType = std::vector<Value>;
 
+    using IPAddress = Poco::Net::IPAddress;
+
+    using IPv4Container = PODArray<UInt32>;
+    using IPv6Container = PaddedPODArray<UInt8>;
+    using IPMaskContainer = PODArray<UInt8>;
+    using RowIdxConstIter = ContainerType<size_t>::const_iterator;
+
+    template <typename T> struct IPContainerToValueType {};
+    template <> struct IPContainerToValueType<IPv4Container> { using type = UInt32; };
```

Comment:
Looks like this code is not valid for gcc: https://stackoverflow.com/a/3052604/6182208
Which causes 
```
../src/Dictionaries/TrieDictionary.h:163:15: error: explicit specialization in non-namespace scope 'class DB::TrieDictionary'
  163 |     template <> struct IPContainerToValueType<IPv4Container> { using type = UInt32; };
```
in ci build

Reply:
I've removed this stuff at all